sqlglot/optimizer/annotate_types.py

```diff
@@ -509,6 +515,23 @@ def _annotate_by_args(
                 ),
             )
 
+        if struct:
+            self._set_type(
+                expression,
+                exp.DataType(
+                    this=exp.DataType.Type.STRUCT,
+                    expressions=[
+                        exp.ColumnDef(
+                            this=exp.to_identifier(expr.alias),
+                            kind=expr.type,
+                            nested=expr.args.get("nested"),
```

Comment:
This doesn't look right, `ColumnDef` has no arg named `nested`. I think we just need to get rid of this.

Reply:
done